### DISCUSSION TITLE
disable honeypots

### DISCUSSION THREAD

**Discussion by alanminana** [telekom-security/tpotce#882] [Q&A]
Hello everyone,

I downloaded the latest version of tpot, and installed the "standard" version, I can not find how to disable the honeypots that I do not use, I read that I had to do systemctl stop tpot, then comment the lines of the honeypots in "/opt/tpot/etc/tpot.yml", and after that perform "systemctl start tpot", but still raising all the same honeypots, also tried to remove them from the web, but nothing happens after rebooting.

greetings

**Comment by Ganesh-403** [comment]
@alanminana 
If you're using a **recent version of T-Pot**, the recommended way to disable honeypots is **not** to stop or delete containers individually. Instead, you should modify the Docker Compose configuration that T-Pot uses.

For current versions, T-Pot provides two supported approaches:

1. **Use one of the predefined compose files** (e.g. `mini.yml`, `sensor.yml`, `standard.yml`), or
2. **Generate a custom compose file** using the built-in `customizer.py` script, which lets you select exactly which honeypots and services you want to run. :contentReference[oaicite:0]{index=0}

For example:

```bash
cd ~/tpotce/compose
python3 customizer.py
```

Then activate the generated configuration:

```bash
systemctl stop tpot
cp docker-compose-custom.yml ~/tpotce/docker-compose.yml
systemctl start tpot
```

The customizer also checks for port conflicts between honeypots before generating the compose file. :contentReference[oaicite:1]{index=1}

---

### If you're using an older T-Pot release (like 21.x/22.x)

The older documentation refers to editing:

```text
/opt/tpot/etc/tpot.yml
```

The recommended workflow was:

```bash
systemctl stop tpot
vi /opt/tpot/etc/tpot.yml

# Test the compose file
docker-compose -f /opt/tpot/etc/tpot.yml up

# Stop the test
docker-compose -f /opt/tpot/etc/tpot.yml down -v

systemctl start tpot
```

Simply commenting out services without validating the compose file can leave dependencies or port mappings that cause T-Pot to recreate containers on startup. :contentReference[oaicite:2]{index=2}

---

### If your changes don't take effect

A few things are worth checking:

- Verify you're editing the **active** compose file for your installed T-Pot version (older releases use `/opt/tpot/etc/tpot.yml`, while newer releases use `~/tpotce/docker-compose.yml`).
- Check whether `docker-compose.yml` is being regenerated or overwritten during startup.
- After restarting T-Pot, confirm which services are actually running:

```bash
docker ps
```

If containers you removed are still starting, it usually means T-Pot is reading a different compose file than the one you modified.

Could you share:
- **Which T-Pot version are you running?**
- Does `docker ps` still show the honeypots you removed after `systemctl start tpot`?

That will help determine whether this is an issue with the compose configuration, the T-Pot version, or the startup process.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
